test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -289,70 +288,90 @@ public void PlattCalibratorOnnxConversionTest()
 
             var initialPipeline = mlContext.Transforms.ReplaceMissingValues("Features").
                 Append(mlContext.Transforms.NormalizeMinMax("Features"));
+            return (mlContext, dataView, estimators, initialPipeline);
+        }
+
+        private void CommonCalibratorOnnxConversionTest(MLContext mlContext, IDataView dataView,
+            List<IEstimator<ITransformer>> estimators, EstimatorChain<NormalizingTransformer> initialPipeline,
+            IEstimator<ITransformer> calibrator, IEstimator<ITransformer> calibratorNonStandard)
+        {
+            // Step 1: Test calibrator with binary prediction trainer
             foreach (var estimator in estimators)
             {
-                var pipeline = initialPipeline.Append(estimator).Append(mlContext.BinaryClassification.Calibrators.Platt());
-                var onnxFileName = $"{estimator}-WithPlattCalibrator.onnx";
-
-                TestPipeline(pipeline, dataView, onnxFileName, new ColumnComparison[] { new ColumnComparison("Score", 3), new ColumnComparison("PredictedLabel"), new ColumnComparison("Probability", 3) });
+                var pipelineEstimators = initialPipeline.Append(estimator).Append(calibrator);
+                var onnxFileName = $"{estimator}-With-{calibrator}.onnx";
+                TestPipeline(pipelineEstimators, dataView, onnxFileName, new ColumnComparison[] { new ColumnComparison("Score", 3), new ColumnComparison("PredictedLabel"), new ColumnComparison("Probability", 3) });
             }
+
+            // Step 2: Test calibrator without any binary prediction trainer
+            IDataView dataSoloCalibrator = mlContext.Data.LoadFromEnumerable(GetCalibratorTestData());
+            var onnxFileNameSoloCalibrator = $"{calibrator}-SoloCalibrator.onnx";
+            TestPipeline(calibrator, dataSoloCalibrator, onnxFileNameSoloCalibrator, new ColumnComparison[] { new ColumnComparison("Probability", 3) });
+
+            // Step 3: Test calibrator with a non-default Score column name and without any binary prediction trainer
+            IDataView dataSoloCalibratorNonStandard = mlContext.Data.LoadFromEnumerable(GetCalibratorTestDataNonStandard());
+            var onnxFileNameSoloCalibratorNonStandard = $"{calibratorNonStandard}-SoloCalibrator-NonStandard.onnx";
+            TestPipeline(calibratorNonStandard, dataSoloCalibratorNonStandard, onnxFileNameSoloCalibratorNonStandard, new ColumnComparison[] { new ColumnComparison("Probability", 3) });
+
             Done();
         }
 
-        class PlattModelInput
+        [Fact]
+        public void PlattCalibratorOnnxConversionTest()
+        {
+            var (mlContext, dataView, estimators, initialPipeline) = GetEstimatorsForOnnxConversionTests();
+            CommonCalibratorOnnxConversionTest(mlContext, dataView, estimators, initialPipeline,
```

Comment:
You can move this call inside `CommonCalibratorOnnxConversionTest`. You will need `mlContext` locally here, but you can simply use `ML.` that is part of every test that inherits from `BaseTestBaseline`. (See `NormalizingEstimatorConversionTests` for an example that doesn't create its own `MLContext`.  You may need to pass that in as a parameter to `CommonCalibratorOnnxConversionTest` though)

Reply:
(Strictly speaking, none of the other tests should need to create their own `MLContext` but that can be cleaned up another day)

---
In reply to: [453127666](https://github.com/dotnet/machinelearning/pull/5289#discussion_r453127666) [](ancestors = 453127666)